File: Data Structures & Algorithms/two-integer-sum/submission-15.py
class Solution:
    # nums[i] + nums[j] == target ==> nums[i] = target - nums[j]
    # 4 ==> is 7-4 in storage(3)? ==> yep, so storag ((3,ind=0), return [0,1]
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        compMap = {}
        for i,val in enumerate(nums):
            comp = target - val
            if comp in compMap:
                return [compMap[comp],i]
            compMap[val] = i
        return []
        

File: Data Structures & Algorithms/duplicate-integer/submission-15.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        bool isDuplicate = false;
        int n = nums.size();
        for(int i = 0; i < n-1; i++){
            for (int j = i+1; j < n; j++){
                if (nums[i] == nums[j]){
                    return not isDuplicate;
                }
            }
        }
        return isDuplicate;
    }
};